platform/ios/src/MGLMapView.mm

```diff
@@ -974,6 +975,9 @@ - (void)setContentInset:(UIEdgeInsets)contentInset animated:(BOOL)animated
     {
         [self didUpdateLocationWithUserTrackingAnimated:animated];
     }
+    
+    // Compass, logo and attribution button constraints needs to be updated.
+    [self setNeedsUpdateConstraints];
```

Comment:
`-layoutSubviews` calls `-adjustContentInset`, which calls `-setContentInset:` and `-setContentInset:animated:`, which calls `-setNeedsUpdateConstraints`. I’m not quite sure, but are there any cases where that’ll cause a feedback loop?